src/theory/arith/nonlinear_extension.cpp

```diff
@@ -1615,6 +1643,125 @@ void NonlinearExtension::getCurrentPiBounds( std::vector< Node >& lemmas ) {
   lemmas.push_back( pi_lem );
 }
 
+Node NonlinearExtension::getApproximateConstant(Node c,
+                                                bool isLower,
+                                                unsigned prec) const
+{
+  Assert(c.isConst());
+  Rational cr = c.getConst<Rational>();
+
+  unsigned lower = 0;
+  unsigned upper = pow(10, prec);
+
+  Rational den = Rational(upper);
```

Comment:
Shouldn't you follow the note at 

https://github.com/CVC4/CVC4/blob/6cdf6fb87a48c04b72722382f43b7fded0b97778/src/util/rational_cln_imp.h#L51

which says that you should avoid creating Rationals from integers with this constructor?